### ISSUE TITLE
failed to lower mrcnn model(https://github.com/ahmedfgad/Mask-RCNN-TF2) with iree-import-tf 

### ISSUE THREAD

**Issue by ArunaKote** [google/iree#6956]
I referrred to https://github.com/ahmedfgad/Mask-RCNN-TF2 and created model with output contains only detection layer.When I run with /data/aruna/iree-latest/iree/integrations/tensorflow/bazel-bin/iree_tf_compiler/iree-import-tf -tf-import-type=savedmodel_v1 -tf-savedmodel-exported-names=serving_default /data/aruna/Mask-RCNN-TF2/detection  --print-ir-after-all &>log.txt -o hlo.mlir

I am getting error as :
 error: The following Tensorflow operations still remain: 
	tf.CropAndResize (count: 4)
	tf.DenseToDenseSetOperation (count: 2)
	tf.NonMaxSuppressionV4 (count: 1)
	tf.SparseToDense (count: 2)
	tf.StridedSlice (count: 21)
	tf.TensorListFromTensor (count: 1)
	tf.TensorListGetItem (count: 1)
	tf.TensorListReserve (count: 1)
	tf.TensorListSetItem (count: 1)
	tf.TensorListStack (count: 1)
	tf.TopKV2 (count: 2)
	tf.Unique (count: 1)
	tf.Where (count: 8)

I was going through the these ops seperately when tried tf.TopKV2 with static inputs I able to lower the op to HLO.

I have attached model file.

[saved_model.zip](https://github.com/google/iree/files/7104927/saved_model.zip)

**Comment by ArunaKote** [comment]
The first part below indicates tf python API and right hand side is the tf ops corresponding to python tf API.
tf.where->tf.where
tf.sets.intersection->tf.DenseToDenseSetOperation
tf.sparse.to_dense->tf.SparseToDense
tf.unique->tf.Unique
tf.image.non_max_suppression->tf.NonMaxSuppressionV4
tf.map_fn->tf.TensorListFromTensor,tf.TensorListGetItem,tf.TensorListReserve,tf.TensorListSetItem ,tf.TensorListStack
tf.nn.top_k->tf.TopKV2
tf.image.crop_and_resize->tf.CropAndResize
x[2:]->tf.StridedSlice
tf.keras.layers.Conv2DTranspose->tf.Conv2DBackpropInput
tf.keras.layers.UpSampling2D(size=(2, 2),interpolation='bilinear')->tf.ResizeBilinear
tf.keras.layers.UpSampling2D(size=(2, 2),interpolation='nearest')->tf.ResizeNearestNeighbor

In this I took seperate test cases and if shapes are static then I able to lower tf.TopKV2,tf.StridedSlice and tf.Conv2DBackpropInput .tf.ResizeBilinear and tf.ResizeNearestNeighbor if it has argument  half_pixel_centers = false I able to lower to MHLO.
But In this case the output tf.where is dynamic and which is given as input to other ops.  In this model half_pixel_centers=true 
these ops are failing while lowering to MHLO using iree-import-tf


Thanks
